### PR TITLE
Changelings can now open their disguise's mail

### LINKED ISSUE #7497: Ghetto Ling Test (rise of the mail cop)
## Reproduction:

If you're disguised as someone as a ling you cant open their mail. Simple as that.

### PR DESCRIPTION
**Pull request by etherware-novice**
## About The Pull Request
Changelings now have the ability to open others mail IF they are disguised as them.

## Why It's Good For The Game
Prevents powergaming by making people try to open their own mail, more gameplay/roleplay opprotunities to steal someone's mail, closes #7497 

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

</details>

## Changelog
:cl:
add: Nanotrasen has weakened the dna lock on mail, allowing changelings to open their disguise's mail.
/:cl:

### CODE REVIEW

**Comment by Penwin0** [comment]
> allowing changelings to open their disguise's mail.

Are you sure?

**Comment by etherware-novice** [comment]
wdym, check the code

**Comment by Penwin0** [comment]
Nevermind, missread

**Comment by itsmeow** [comment]
theoretically, a changeling could take a renaming potion to open any mail. Perhaps this should check something else, do changelings have a unique disguised variable or something? Check that instead of name.

**Comment by etherware-novice** [comment]
uhh i havent played ling or looked at the ling code yet what variable is that?? like dna?

**Comment by FortSumter** [comment]
Imagine trying to metagame changelings like this.

On the topic of changelings, they really need some RP fun abilities besides spawning tumor babies and turning everyone into moffs

**Comment by etherware-novice** [comment]
thats not my job here

also i thought they had an ability to turn everyone into one person?

File: code/game/objects/items/mail.dm [hunk 1]
```diff
@@ -155,8 +155,13 @@
...
 		if(recipient && recipient != user?.mind)
-			to_chat(user, "<span class='notice'>You can't open somebody else's mail! That's <em>immoral</em>!</span>")
-			return
+			if(is_changeling(user))
+				if(user.real_name != recipient.name)
+					to_chat(user, "<span class='warning'>We must keep our disguise intact.</span>")  // cuz your disguise cant open the mail so you shouldnt either
+					return
+			else
+				to_chat(user, "<span class='notice'>You can't open somebody else's mail! That's <em>immoral</em>!</span>")
+				return
```
Comment (PowerfulBacon) [inline code/game/objects/items/mail.dm]:
```suggestion
			if(!is_changeling(user))
				to_chat(user, "<span class='notice'>You can't open somebody else's mail! That's <em>immoral</em>!</span>")
				return
			if(user.real_name != recipient.name)
				to_chat(user, "<span class='warning'>We must keep our disguise intact.</span>")  // cuz your disguise cant open the mail so you shouldnt either
				return
```

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**